### DISCUSSION TITLE
Could not resolve "pg-native"

### DISCUSSION THREAD

**Discussion by pajasry** [mikro-orm/mikro-orm#2834] [Q&A]
**Describe the bug**
After successful installation the server shows me an error ```node_modules/pg/lib/native/client.js:4:21: ERROR: Could not resolve "pg-native"```. I tried to install pg-native manually, but without success.

**Stack trace**
```
node_modules/pg/lib/native/client.js:4:21: ERROR: Could not resolve "pg-native"
```

**Expected behavior**
I can use mikro-orm.

**Additional context**
Even though I don't have pg-native I can use micro-orm/cli to create table and migrations

**Versions**

| Dependency | Version |
| - | - |
| node | 16.13.0 |
| typescript | 4.5.5 |
| mikro-orm | 5.5 |
| postgres | 14.2 |

**Comment by B4nan** [comment]
MikroORM does not explicitly support pg-native, it uses just the `pg` library, via knex. I am not sure if you are trying to use just the pg driver, or you actually want to use pg-native. I can assure you there is no need to have pg-native installed (and to support it I believe we would need a new driver, given knex also has a separate dialect for it).

Moving to discussions as there is no reproduction. How does your ORM config look like? Maybe you are doing something with `driverOptions`? If you want help, I need details. 

Note that you should not install `pg`, nor `pg-native` nor `knex` - those are all transitive dependencies, you just install `core` package and the driver.

Maybe you are using some non standard way to compile TS or bundle your app? I can imagine this if you try to use webpack.

**Reply by pajasry** [reply]
I apologize for the lack of information. I didn't know what was essential. Could you tell me more about the compilation because I think there is a solution to my problem as i had faced problems with missing other driver (i must install all of them to get working mikor-orm). I am aware of dependencies, that i should install and i will be the happiest developer under the sun if i can go with it ;)

My mikro-orm.config.ts:
```
MikroORM.init<PostgreSqlDriver>({
    metadataProvider: ReflectMetadataProvider,
    entities: [Organization],
    entitiesTs: ['./app/src/entities'],
    highlighter: new SqlHighlighter(),
    type: 'postgresql',
})
```
package.json
```
{
  "name": "remix-app-template",
  "private": true,
  "description": "",
  "license": "",
  "sideEffects": false,
  "scripts": {
    "build": "cross-env NODE_ENV=production remix build",
    "dev": "cross-env NODE_ENV=development remix dev",
    "postinstall": "remix setup node",
    "start": "cross-env NODE_ENV=production remix-serve build"
  },
  "dependencies": {
    "@mikro-orm/better-sqlite": "^5.0.5",
    "@mikro-orm/core": "^5.0.5",
    "@mikro-orm/entity-generator": "^5.0.5",
    "@mikro-orm/mariadb": "^5.0.5",
    "@mikro-orm/mongodb": "^5.0.5",
    "@mikro-orm/mysql": "^5.0.5",
    "@mikro-orm/postgresql": "^5.0.5",
    "@mikro-orm/reflection": "^5.0.5",
    "@mikro-orm/seeder": "^5.0.5",
    "@mikro-orm/sql-highlighter": "^1.0.1",
    "@mikro-orm/sqlite": "^5.0.5",
    "@remix-run/react": "^1.2.2",
    "@remix-run/serve": "^1.2.2",
    "cross-env": "^7.0.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "remix": "^1.2.2",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@mikro-orm/cli": "^5.0.5",
    "@mikro-orm/migrations": "^5.0.5",
    "@remix-run/dev": "^1.2.2",
    "@types/react": "^17.0.24",
    "@types/react-dom": "^17.0.9",
    "@types/uuid": "^8.3.4",
    "ts-node": "^10.5.0",
    "typescript": "^4.5.5"
  },
  "mikro-orm": {
    "useTsNode": true,
    "configPaths": [
      "./app/src/mikro-orm.config.ts"
    ]
  },
  "engines": {
    "node": ">=14"
  }
}
```

**Reply by B4nan** [reply]
If you need to install all drivers, its most probably exactly this. Idk how remix works under the hood, but if its using webpack, check this:

https://mikro-orm.io/docs/deployment#webpack-configuration

Also you don't need `entitiesTs` here, that's needed only for folder based discovery. If you want to use folder based discovery, both `entities` and `entitiesTs` would need to do it, your current setup is a mixture of those two approaches. 

So either just `entities: [Foo, Bar]` or you have both `entities: ['dist/entities/*'], entitiesTs: ['src/entities/*']`.

**Reply by B4nan** [reply]
This is probably related: https://github.com/remix-run/remix/issues/1770

**Reply by B4nan** [reply]
If you can share your project, I'll be happy to take a look, would be good to have an official remix example. There was someone on slack trying out remix, idk about the state of that project, but its here so maybe you find it useful: https://github.com/Industrial/test-remix-vercel

Looking at [this](https://github.com/Industrial/test-remix-vercel/blob/main/remix.config.js#L10-L17), it might be exactly what you need.

**Reply by pajasry** [reply]
Thank you for your time and effort. I dive deeper to remix and its compilation. Problem is in missing external options. I had also bounced to [this PR](https://github.com/remix-run/remix/pull/1841) which confirmed my assumptions. I will take a look on it and eventualy contribute with example.
